backend/media/src/main/java/co/airy/core/media/services/MediaUpload.java

```diff
@@ -14,22 +19,30 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLConnection;
+import java.util.List;
 
 @Service
 @EnableRetry
 public class MediaUpload {
```

Comment:
So the cool thing about healthchecks in Spring is that _any_ component can signal its healthy state. For instance `MediaUpload` can also implement the `HealthIndicator` interface. This way we don't need to query the `connectionStatus` variable from the `MediaController`

Sorry, I forgot to mention this. The rest of the code looks great 🙏

Reply:
Ok no worries, I moved the Heath check from `MediaController` to `MediaUpload`. Because there is no logic on the `MediaController` Heath implementation. I don't think it is necessary to have one on there